alternatives.c

```diff
@@ -1304,13 +1328,18 @@ int main(int argc, const char **argv) {
     struct stat sb;
     struct linkSet newSet = {NULL, NULL, NULL};
 
+    if (canUseAlternativeAdminDirUnderOSTree()) {
+        stateDir = "/etc/alternatives.admindir";
+    }
+
     setlocale(LC_ALL, "");
     bindtextdomain("chkconfig", "/usr/share/locale");
     textdomain("chkconfig");
```

Comment:
Since `canUseAlternativeAdminDirUnderOSTree()` is invoked before `setlocale()`, the diagnostic message will not be translated. Also, if `--admindir` option is used, `canUseAlternativeAdminDirUnderOSTree()` should not be invoked at all.  You can even create a setup where even `--help` and `--version` options will not work after this change.

Reply:
Will update, thanks for catching!